.github/workflows/test.yml

```diff
@@ -0,0 +1,31 @@
+name: Test
+
+on:
+  push:
+    branches: [ master, gocardless ]
+  pull_request:
+    branches: [ master, gocardless ]
+
+concurrency:
+  # running pipeline per workflow per PR
+  group: ${{ github.head_ref || github.run_id }}-${{ github.workflow }}
+  # Running a new pipeline will cancel any running pipelines that belong to the above group
+  cancel-in-progress: true
+
+jobs:
+  test:
+    runs-on: ubuntu-22.04
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby-version: [ '3.1' ]
```

Comment:
do you think it's valuable adding here all the active ruby versions?

Reply:
Maybe, later on, when we integrate it back upstream. Don't care too much about it right now